device-connectors/src/testflinger_device_connectors/devices/zapper_kvm/__init__.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (C) 2024 Canonical
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Zapper Connector for KVM provisioning."""
+
+import os
+from typing import Any, Dict, Tuple
+
+from testflinger_device_connectors.devices.zapper import ZapperConnector
+
+
+class DeviceConnector(ZapperConnector):
+    """Tool for provisioning baremetal with a given image."""
+
+    PROVISION_METHOD = "ProvisioningKVM"
+
+    def _get_autoinstall_conf(self):
+        """Prepare autoinstall-related configuration."""
+        autoinstall_conf = {}
+
+        autoinstall_conf["storage_layout"] = self.job_data["provision_data"][
+            "storage_layout"
+        ]
+
+        autoinstall_conf["storage_password"] = self.job_data[
+            "provision_data"
+        ].get("storage_password")
```

Comment:
I guess this part of code can be simplified the following way:

```suggestion
        autoinstall_conf = {
            "storage_layout": self.job_data["provision_data"]["storage_layout"],
            "storage_password": self.job_data["provision_data"].get("storage_password")
        }
```